packages/web-api/src/WebClient.ts

```diff
@@ -471,7 +471,8 @@ export class WebClient extends EventEmitter<WebClientEvent> {
     open: (this.apiCall.bind(this, 'views.open')) as Method<methods.ViewsOpenArguments>,
     publish: (this.apiCall.bind(this, 'views.publish')) as Method<methods.ViewsPublishArguments>,
     push: (this.apiCall.bind(this, 'views.push')) as Method<methods.ViewsPushArguments>,
-    update: (this.apiCall.bind(this, 'views.update')) as Method<methods.ViewsUpdateArguments>,
+    update: (this.apiCall.bind(this, 'views.update')) as
```

Comment:
i'm a little concerned that this is a break from the existing pattern. there are no other methods which use a union type here.

if we want to do this, i would rather seem a `type ViewUpdateArguements = ViewsUpdateArgumentsWithViewId | ViewsUpdateArgumentsWithExternalId` in `methods.ts` so that the complexity of underlying shapes stays out of `WebClient`.

Reply:
That's a good point. Doing a quick scan, I don't see any union type methods like your suggested `ViewUpdateArugments` in the `mehods.ts` file. I know though that it is a pattern we have used in `bolt`.